### PR TITLE
An initial attempt at a parser and printer for the boolean language

### PR DESCRIPTION
**Pull request by kwxm**
This is very simplistic and probably not very Haskelly.  I'm not totally convinced that the printer is correct because the parser isn't finished yet and it's tricky to test things without it.  I know we should probably use a library to do the pretty-printing, but in this case it was quicker just to do it by hand.

I've added Show instances to the Expr type to make it easier to see what's going on in ghci.

### CODE REVIEW

**Comment by kwxm** [comment]
I also added Eq to Var to make lookup easier.  It might also be helpful to have an official way of getting the basic name out of a Var rather than using _varName

File: tiny-lang.cabal [hunk 1]
```diff
@@ -18,7 +18,12 @@ library
   exposed-modules:     TinyLang.Prelude,
                        TinyLang.Var
                        TinyLang.Boolean.Core
+                       TinyLang.Boolean.Evaluator
+                       TinyLang.Boolean.Environment
+                       TinyLang.Boolean.Parser
```
Comment (effectfully) [inline tiny-lang.cabal:23]: The actual file doesn't seem to be added.

**Review by effectfully — APPROVED** [review]
Thank you!

File: tiny-lang.cabal (hunk 1 above)
Comment (kwxm) [reply tiny-lang.cabal:23]: Parser.hs? Do you want me to add that? It doesn't quite work at the moment and I'll probably rewrite it using the Expr library.  Maybe you should just take Parser out of the cabal file for now.

Comment (effectfully) [reply tiny-lang.cabal:23]: Yeah, please take it out, otherwise the code will not build.

Comment (effectfully) [reply tiny-lang.cabal:23]: Ah, sorry, misread as "Maybe I should just take Parser". I'll take it out.

**Pull request merged**